python-lib/googlesheets_common.py

```diff
@@ -0,0 +1,12 @@
+def get_credentials(config):
+    auth_type = config.get("auth_type", "credentials")
+
+    if auth_type == "credentials":
+        # Legacy mode
+        credentials = config.get("credentials")
+        return credentials
+    elif auth_type == "service-account":
+        preset_credentials = config.get("preset_credentials", {})
+        credentials = preset_credentials.get("credentials", None)
+        return credentials
+    return None
```

Comment:
There are now two methods around call get_credentials() and it's enough to be a bit confusing

One is in the other file, I googlesheets.py, and it could probably be make "private" with a _ prefix and renamed to something like "get_sa_credentials()". 

The one here in googlesheets_common.py could probably be renamed to extract_credentials(config).

Or maybe I don't get it entirely but that's my suggestion based on my understanding of it!